### PR TITLE
Standardize AWS Athena naming

### PR DESCRIPTION
**Pull request by ferruzzi**
Part of https://github.com/apache/airflow/issues/20296

Also adjusted some tests to use `poke({})` and `execute({})` instead of `poke(None)` and `execute(None)` as the IDE was throwing type hint warnings on those.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by ferruzzi** [comment]
Missed renaming the hooks in the sensor tests, update coming.

File: airflow/providers/amazon/aws/hooks/athena.py [hunk 1]
```diff
@@ -25,7 +26,17 @@
...
 
-class AWSAthenaHook(AwsBaseHook):
+class AWSAthenaHook:
+    """Deprecated Hook"""
+
+    warnings.warn(
+        "This hook is deprecated. Please use `airflow.providers.amazon.aws.hooks.athena.AthenaHook`.",
+        DeprecationWarning,
+        stacklevel=2,
+    )
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/athena.py]: This would make this warning unconditionally appear (even if this class is never actually used), and existing code to all break. You should

1. Make `AWSAthenaHook` subclass `AthenaHook` (instead of the other way around)
2. Move the deprecation warning to `__init__`.

File: airflow/providers/amazon/aws/operators/athena.py [hunk 2]
```diff
@@ -25,16 +26,27 @@
...
-class AWSAthenaOperator(BaseOperator):
+class AWSAthenaOperator:
+    """Deprecated Operator"""
+
+    warnings.warn(
+        "This operator is deprecated. Please use "
+        "`airflow.providers.amazon.aws.operators.athena.AthenaOperator`.",
+        DeprecationWarning,
+        stacklevel=2,
+    )
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/operators/athena.py]: Same as above.

**Review by uranusjr — DISMISSED** [review]

File: airflow/providers/amazon/aws/hooks/athena.py [hunk 3]
```diff
@@ -260,3 +261,16 @@ def stop_query(self, query_execution_id: str) -> Dict:
...
+    """
+    This hook is deprecated.
+    Please use :class:`airflow.providers.amazon.aws.hooks.athena.AthenaHook`.
+    """
+
+    warnings.warn(
+        "This hook is deprecated. Please use `airflow.providers.amazon.aws.hooks.athena.AthenaHook`.",
+        DeprecationWarning,
+        stacklevel=2,
+    )
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/athena.py]:
```suggestion
class AWSAthenaHook(AthenaHook):
    """
    This hook is deprecated.
    Please use :class:`airflow.providers.amazon.aws.hooks.athena.AthenaHook`.
    """

    def __init__(self, *args, **kwargs):
        warnings.warn(
            "This hook is deprecated. Please use `airflow.providers.amazon.aws.hooks.athena.AthenaHook`.",
            DeprecationWarning,
            stacklevel=2,
        )
        super().__init__(*args, **kwargs)
```

Having the warning directly in the class body would cause the message to be emitted _on import time_, which is not correct.

Comment (ferruzzi) [reply airflow/providers/amazon/aws/hooks/athena.py]: Sorry, just saw these comments after I made those changes.  I'll move those into inits.

[Done]

File: setup.cfg [hunk 4]
```diff
@@ -45,6 +45,7 @@ license_files =
    licenses/LICENSE-moment-strftime.txt
    licenses/LICENSE-moment.txt
    licenses/LICENSE-normalize.txt
+   licenses/LICENSES-ui.txt
```
Comment (uranusjr) [inline setup.cfg]: This seems unrelated?

Comment (ferruzzi) [reply setup.cfg]: the pre-commits keep adding that??

Comment (uranusjr) [reply setup.cfg]: Hmm that doesn’t make sense, there’s an identical line just below and pre-commit should catch that. Something is probably broken somewhere. Could you temporarily disable pre-commit to remove that line and push to see if CI would let it pass?

Comment (eladkal) [reply setup.cfg]: yep this change should be removed.
this PR should not change `setup.cfg`

Comment (ferruzzi) [reply setup.cfg]: Change reverted and pushed.  

My pre-commit hooks also started failing a whole bunch of mypy tests today when I pulled the latest main down, so I may need to unbork something on my environment.  Sorry I missed that.

Comment (ferruzzi) [reply setup.cfg]: For context, I just now rebased main and ran the pre-commits locally using `pre-commit run --all-files` on it without any of my code changes and it did the same:

```
Update setup.cfg file with all licenses..................................................Failed
- hook id: update-setup-cfg-file
- files were modified by this hook
```

File: tests/providers/amazon/aws/sensors/test_athena.py [hunk 5]
```diff
@@ -36,26 +36,26 @@ def setUp(self):
...
-            self.sensor.poke(None)
+            self.sensor.poke({})
         assert 'Athena sensor failed' in str(ctx.value)
 
-    @mock.patch.object(AWSAthenaHook, 'poll_query_status', side_effect=("CANCELLED",))
+    @mock.patch.object(AthenaHook, 'poll_query_status', side_effect=("CANCELLED",))
     def test_poke_cancelled(self, mock_poll_query_status):
         with pytest.raises(AirflowException) as ctx:
-            self.sensor.poke(None)
+            self.sensor.poke({})
```
Comment (eladkal) [inline tests/providers/amazon/aws/sensors/test_athena.py:60]: This is not really related to the PR description?
I guess you fix something else here on the way? If so just add some description about it in the PR

**Review by eladkal — APPROVED** [review]
LGTM

@uranusjr any further comments?

File: tests/providers/amazon/aws/sensors/test_athena.py (hunk 5 above)
Comment (ferruzzi) [reply tests/providers/amazon/aws/sensors/test_athena.py:60]: Yeah, sorry.   I'm really bad at that; I fix little things when I see them because I know I'll forget or won't get to it later.  IDE was throwing a type hint warning there, expecting a dict and got NoneType.  I'll add a comment to the description.

**Comment by ferruzzi** [comment]
Looks like I'll have to rebase to clear some merge conflicts.  I'll take care of that tomorrow/Thursday.

**Review by eladkal — APPROVED** [review]
LGTM

If there will be more comments we'll address them in a followup PR.
